src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -146,11 +149,47 @@ public DateTimeKind GetCorrespondingKind(TimeZoneInfo? timeZone)
         /// </summary>
         public bool HasIanaId { get; }
 
-        public string DisplayName => _displayName ?? string.Empty;
+        public string DisplayName
+        {
+            get
+            {
+                if (_displayName == null && !_displayNameSet)
```

Comment:
>_displayName == null && !_displayNameSet [](http://example.com/codeflow?start=20&length=40)

Isn't enough checking _displayNameSet only? in other word, is it possible _displayName is not null and in same time _displayNameSet is false?

Reply:
I think you're right, I haven't missed any other cases where `_displayName` is set to `null` besides `CreateCustomTimeZone`